tools/mafft/mafft.xml

```diff
@@ -17,189 +17,300 @@
-        mafft
-        $cond_flavour.dist_flavour.distance_method
-        #if $cond_flavour.dist_flavour.distance_method == '--6merpair'
-          --retree $cond_flavour.dist_flavour.retree
-          $cond_flavour.dist_flavour.distance_method.usetree.parttree
+      ## sequence_counter
+  #set sequence_count = 0
+  #if $concatenate.concatenate_options == "single"   
+    #for $data in $concatenate.datasets:
+      #set sequence_count += int($data.inputs.metadata.sequences)
+    #end for 
+  #elif $concatenate.concatenate_options == "row"
+    #for $data in $concatenate.datasets:
+      #for $input in $data.inputs:
+        #set sequence_count += int($input.metadata.sequences)
+      #end for  
+    #end for
+  #end if
 
-          #if $cond_flavour.dist_flavour.distance_method.usetree.parttree==--parttree
-            $cond_flavour.dist_flavour.distance_method.usetree.treedistance
-            $cond_flavour.dist_flavour.distance_method.usetree.partsize
-            $cond_flavour.dist_flavour.distance_method.usetree.groupsize
-          #end if
+  ## FASTA36 
+  #if $cond_flavour.flavourType == "custom" 
+    #if $cond_flavour.guidetree.guidetree_generation == "original"  
+      #if $cond_flavour.guidetree.dist_flavour.distance_method == "--fastapair"
+        export FASTA_4_MAFFT=`which fasta36` &&
+      #end if
+    #elif $cond_flavour.guidetree.guidetree_generation == "parttree"
+      #if $cond_flavour.guidetree.parttree_selection.parttree_option == "--fastaparttree"
+        export FASTA_4_MAFFT=`which fasta36` &&
+      #end if
+    #end if
+  #end if
 
-        #elif $cond_flavour.dist_flavour.distance_method == '--globalpair'
-          --weighti $cond_flavour.dist_flavour.weighti
-        #elif $cond_flavour.dist_flavour.distance_method == '--localpair'
-          --weighti $cond_flavour.dist_flavour.weighti
-          --lop $cond_flavour.dist_flavour.lop
-          --lep $cond_flavour.dist_flavour.lep
-          --lexp $cond_flavour.dist_flavour.lexp
-        #elif $cond_flavour.dist_flavour.distance_method == '--genafpair'
-          --weighti $cond_flavour.dist_flavour.weighti
-          --lop $cond_flavour.dist_flavour.lop
-          --lep $cond_flavour.dist_flavour.lep
-          --lexp $cond_flavour.dist_flavour.lexp
-          --LOP $cond_flavour.dist_flavour.skipLOP
-          --EXP $cond_flavour.dist_flavour.skipEXP 1
-        #elif $cond_flavour.dist_flavour.distance_method == '--fastapair'
-          --weighti $cond_flavour.dist_flavour.weighti
+    ## groupsize warning
+  #if $cond_flavour.flavourType == "custom"
+    #if $cond_flavour.guidetree.guidetree_generation == "parttree"
+      echo $cond_flavour.guidetree.parttree_selection.groupsize &&
+      #if $cond_flavour.guidetree.parttree_selection.groupsize == "None"
+        int($cond_flavour.guidetree.parttree_selection.groupsize) = $sequence_count 
+      #elif $cond_flavour.guidetree.parttree_selection.groupsize > $sequence_count

```

Comment:
Although the error:
```
'>' not supported between instances of 'NoneType' and 'int'
```
looks like it's coming from here, you can completely remove the whole section and when you don't enter anything for the optional integer param still get it in exactly the same form.

I think this comes from an internal attempt to compare the `None` to the "min" and "max" values of the param =>
There's just no way to combine min, max and optional="true" for an int param.

As discussed try to use -1 instead and explain it in the param help.

Reply:
-1 isn't working. Then MAFFT gives an error like: Specify groupsize!
I think it is an internal function of MAFFT, if one adding a larger number than input sequences, then it shows -1 but it takes one more than the number of input sequences.

I'm just wondering, because there was a moment, when it worked with the comparison.

Reply:
I can remove the section but then we dont have a warning in die stoud. I can simply mention it somewhere else?